pump/storage/storage.go

```diff
@@ -480,6 +480,7 @@ func (a *Append) resolve(startTS int64) bool {
 	maxSecond := oracle.ExtractPhysical(uint64(latestTS)) / int64(time.Second/time.Millisecond)
 	elapseSecond := maxSecond - startSecond
 
+	// `GetTxnStatus` will not cleanup any locks now, but we still keep this logic and only `GetTxnStatus` after `maxTxnTimeoutSecond`.
```

Comment:
`GetTxnStatus` may cleanup expired locks whose `lock.startTS` + `lock.ttl` < `currentTS`. The returned `TxnStatus` is more complex than before we may refer the TiDB usage like https://github.com/pingcap/tidb/blob/master/store/tikv/lock_resolver.go#L361, or review the code in tikv https://github.com/tikv/tikv/blob/release-4.0/src/storage/mvcc/txn.rs#L905.

Reply:
I want to express "do not abort txn now", 😹 .

"cleanup expired locks" will not take any side effects, is it right?

Reply:
"cleanup expired locks" is the way to abort the transaction. If the lock is rolled back, the transaction coult not commit any more.